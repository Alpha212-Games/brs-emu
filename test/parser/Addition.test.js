const Parser = require("../../lib/parser");
const Expr = require("../../lib/parser/Expression");
const { Lexeme } = require("../../lib/Lexeme");
const OrbsError = require("../../lib/Error");

const { token, EOF } = require("./ParserTests");

describe("parser", () => {
    afterEach(() => OrbsError.reset());
    describe("addition expressions", () => {
        it("parses left-associative addition chains", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 1),
                token(Lexeme.Plus),
                token(Lexeme.Integer, 2),
                token(Lexeme.Plus),
                token(Lexeme.Integer, 3),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses left-associative subtraction chains", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 1),
                token(Lexeme.Minus),
                token(Lexeme.Integer, 2),
                token(Lexeme.Minus),
                token(Lexeme.Integer, 1),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });
});