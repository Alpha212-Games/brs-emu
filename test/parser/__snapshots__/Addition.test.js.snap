// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser addition expressions parses left-associative addition chains 1`] = `
Binary {
  "left": Binary {
    "left": Literal {
      "value": 1,
    },
    "right": Literal {
      "value": 2,
    },
    "token": Object {
      "kind": 8,
      "line": 1,
      "literal": undefined,
    },
  },
  "right": Literal {
    "value": 3,
  },
  "token": Object {
    "kind": 8,
    "line": 1,
    "literal": undefined,
  },
}
`;

exports[`parser addition expressions parses left-associative subtraction chains 1`] = `
Binary {
  "left": Binary {
    "left": Literal {
      "value": 1,
    },
    "right": Literal {
      "value": 2,
    },
    "token": Object {
      "kind": 7,
      "line": 1,
      "literal": undefined,
    },
  },
  "right": Literal {
    "value": 1,
  },
  "token": Object {
    "kind": 7,
    "line": 1,
    "literal": undefined,
  },
}
`;
